pkg/render/monitor/monitor.go

```diff
@@ -981,10 +987,33 @@ func (mc *monitorComponent) operatorRole() *rbacv1.Role {
 			},
 		},
 	}
+
+	secretRole := &rbacv1.Role{
+		TypeMeta:   metav1.TypeMeta{Kind: "Role", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: CalicoPrometheusOperatorSecret, Namespace: common.TigeraPrometheusNamespace},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{
+					"secrets",
+				},
+				Verbs: []string{
```

Comment:
Listed specific verbs instead of using * for clearer visibility. This excludes the PATCH and DELETE verbs. Let me know if these verbs should be included. @rene-dekker